core/controllers/acl_decorators.py

```diff
@@ -3572,3 +3573,54 @@ def test_can_update_suggestion(
 
     test_can_update_suggestion.__wrapped__ = True
     return test_can_update_suggestion
+
+
+def is_from_oppia_android(handler):
+    """Decorator to check whether the request was sent from Oppia Android.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function.
+    """
+
+    def test_is_from_oppia_android(self, **kwargs):
+        """Checks whether the request was sent from Oppia Android.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            UnauthorizedUserException. If incoming request is not from a valid
+                Oppia Android request.
+        """
+        headers = self.request.headers
+        api_key = headers['api_key']
+        app_package_name = headers['app_package_name']
+        app_version_name = headers['app_version_name']
+        app_version_code = headers['app_version_code']
+
+        version_name_matches = (
+            android_validation_constants.APP_VERSION_WITH_HASH_REGEXP.match(
+                app_version_name))
+        version_code_is_positive_int = app_version_code.isdigit() and (
+            int(app_version_code) > 0)
+        if (
+                api_key != android_validation_constants.ANDROID_API_KEY or
+                app_package_name != (
+                    android_validation_constants.ANDROID_APP_PACKAGE_NAME) or
+                not version_name_matches or
+                not version_code_is_positive_int):
```

Comment:
This check looks correct--I think we can go ahead and remove the commented out code below this.

Also, are there tests for verifying that this check is working as expected? If not, can you please add them?

Reply:
Done -- tests are largely the same as the `incoming_app_feedback_report` tests but checks each header individually.